core/templates/pages/exploration-player-page/services/question-player-engine.service.spec.ts

```diff
@@ -768,48 +774,42 @@ describe('Question player engine service ', () => {
       spyOn(focusManagerService, 'generateFocusLabel')
         .and.returnValue('focusLabel');
 
+      // We are using a stub backend dict which consists of three questions.
       questionPlayerEngineService.init(
         multipleQuestionsBackendDict, successHandler, failHandler);
 
       let createNewCardSpy = spyOn(
         stateCardObjectFactory, 'createNewCard').and.returnValue(sampleCard);
 
+      expect(createNewCardSpy).toHaveBeenCalledTimes(0);
+
+      // Submitting answer to the first question.
       questionPlayerEngineService.submitAnswer(
         answer, interactionRulesService, successCallback);
 
-      expect(createNewCardSpy).toHaveBeenCalled();
-    });
-
-    it('should not create next question if the existing ' +
-      'question is the last one', () => {
-      let successCallback = jasmine.createSpy('success');
-      let successHandler = jasmine.createSpy('success');
-      let failHandler = jasmine.createSpy('fail');
-      let answer = 'answer';
-      let interactionRulesService: InteractionRulesService;
-      let answerClassificationResult = new AnswerClassificationResult(
-        outcomeObjectFactory
-          .createNew('default', '', '', []), 1, 0, 'default_outcome'
-      );
-      answerClassificationResult.outcome.labelledAsCorrect = true;
-
-      spyOn(answerClassificationService, 'getMatchingClassificationResult')
-        .and.returnValue(answerClassificationResult);
-      spyOn(expressionInterpolationService, 'processHtml')
-        .and.callFake((html, envs) => html);
-      spyOn(focusManagerService, 'generateFocusLabel')
-        .and.returnValue('focusLabel');
+      expect(
+        questionPlayerEngineService.getCurrentQuestionId()).toBe('questionId1');
+      expect(createNewCardSpy).toHaveBeenCalledTimes(1);
 
-      questionPlayerEngineService.init(
-        [singleQuestionBackendDict], successHandler, failHandler);
+      questionPlayerEngineService.recordNewCardAdded();
+      // Submitting answer to the second question.
+      questionPlayerEngineService.submitAnswer(
+        answer, interactionRulesService, successCallback);
 
-      let createNewCardSpy = spyOn(
-        stateCardObjectFactory, 'createNewCard').and.returnValue(null);
+      expect(
+        questionPlayerEngineService.getCurrentQuestionId()).toBe('questionId2');
+      expect(createNewCardSpy).toHaveBeenCalledTimes(2);
 
+      questionPlayerEngineService.recordNewCardAdded();
+      // Submitting answer to the last question.
       questionPlayerEngineService.submitAnswer(
         answer, interactionRulesService, successCallback);
 
-      expect(createNewCardSpy).not.toHaveBeenCalled();
+      expect(
+        questionPlayerEngineService.getCurrentQuestionId()).toBe('questionId3');
+      // Please note that after submitting answer to the final question new
```

Comment:
... final question, a new card was not ...

Reply:
Has this been addressed?

Reply:
done